src/Lang/CorePriv/Effect.ml

```diff
@@ -34,6 +34,10 @@ let prog_effect = TEffJoin(io, nterm)
 (** Check if the effect is pure *)
 let rec is_pure eff =
   match eff with
+  | TUVar _ ->
+    InterpLib.Error.report ~cls:FatalError
```

Comment:
It is rather an internal error (see `InterpLib.InternalError` module) than a regular error. It should never happen, and once it is raised it is information to developers that an internal invariant were broken. You can use also `failwith` function for raising internal errors.